tasks.c

```diff
@@ -906,8 +910,8 @@ static void prvYieldForTask( TCB_t * pxTCB,
                     else if( pxTCB == pxCurrentTCBs[ xCoreID ] )
                     {
                         configASSERT( ( pxTCB->xTaskRunState == xCoreID ) || ( pxTCB->xTaskRunState == taskTASK_YIELDING ) );
-                        #if ( configUSE_CORE_EXCLUSION == 1 )
-                            if( ( pxTCB->uxCoreExclude & ( 1 << xCoreID ) ) == 0 )
+                        #if ( configUSE_CORE_AFFINITY == 1 )
+                            if( ( pxTCB->uxCoreAffinityMask & ( 1 << xCoreID ) ) != 0 )
```

Comment:
Suggestion: Separate #if / #else / #endif sections would make this logic more clear.